### ISSUE TITLE
Remove extra fields from search results before wakeup

### ISSUE THREAD

**Issue by elbrujohalcon** [inaka/sumo_db#206]
I have a very naïve `sumo_wakeup` function:
```erlang
-spec sumo_wakeup(sumo:doc()) -> terminal().
sumo_wakeup(Terminal) -> Terminal.
```
I'm using `sumo_store_riak` and then…
```erlang
1> sumo:find_by(things, [{a_field,<<"value">>}]).
[#{'_yz_id' => <<"1*bc_terminals_maps*bc_terminals*8JBnjO9cnGv1peo0kPSWNhoMKvD|b*40">>,
   '_yz_rb' => <<"bc_terminals">>,
   '_yz_rk' => <<"8JBnjO9cnGv1peo0kPSWNhoMKvD|b">>,
   '_yz_rt' => <<"bc_terminals_maps">>,
   created_at => {{2015,10,23},{13,5,9}},
   description => <<"c">>,
   id => <<"the-id">>,
   a_field => <<"value">>,
   score => <<"3.06852820000000026113e-01">>,
   terminal_id => <<"b">>,
   updated_at => {{2015,10,23},{13,5,9}}}]
2> sumo:find_all(things).
[#{created_at => {{2015,10,23},{13,5,9}},
   description => <<"c">>,
   id => <<"the-id">>,
   a_field => <<"value">>,
   terminal_id => <<"b">>,
   updated_at => {{2015,10,23},{13,5,9}}}]
```

**Comment by elbrujohalcon** [comment]
Riak-ext is already doing this, so it should be just a matter of bringing it in

**Comment by elbrujohalcon** [comment]
Fixed in #213

**Comment by harenson** [comment]
Already fixed, tested using a naïve `sumo_wakeup/1`:
```erlang
-spec sumo_wakeup(sumo:doc()) -> person().
sumo_wakeup(Person) -> Person.
```

Using `sumo_store_riak`:
```erlang
1> sumo:find_by(sumo_test_people_riak, [{name, <<"Alan">>}]).
[#{address => <<"Computer St.">>,
   age => 102,
   birthdate => {2016,2,24},
   created_at => {{2016,2,24},{14,23,24}},
   description => undefined,
   height => undefined,
   id => <<"IKv70F4QWXW9Dv1AfwvRMXE8BCe">>,
   last_name => <<"Turing">>,
   name => <<"Alan">>,
   profile_image => undefined}]
```
